.github/workflows/PR_summary.yml

```diff
@@ -46,10 +46,29 @@ jobs:
         sudo apt-get install -y jq
         # If you have additional dependencies, install them here
 
-    - name: Get changed files
+    - name: Get changed and removed/renamed files
       run: |
         git fetch origin ${{ github.base_ref }}  # fetch the base branch
-        git diff --name-only origin/${{ github.base_ref }}... > changed_files.txt  # get the list of changed files
+        # Get the list of all changed files.
+        git diff --name-only origin/${{ github.base_ref }}... > changed_files.txt
+        # Get all files which were removed or renamed.
+        movedFiles=$(git diff --name-only --diff-filter DR origin/${{ github.base_ref }}...)
```

Comment:
Maybe, instead of defining the variable, you could do `git diff ... | tee filename`, which should print to stdout as well as creating the file.  This also probably prevents issues with filenames that might have embedded commands.